File: src/ProjectR/content/scripts/spells/SecretTechnique76.lua
function GetTargetType()
    return Single
end

function GetName()
return "Secret Technique #76"
end

function GetDescription()
return "A vicious strike that attempts to weaken the enemy.\nChance to steal stat buffs from target."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .76
end

function GetDelay()
  return .3
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return { ARC }
end

function SpellEffect()
  local damage = (3.75 * aAD * (aARC/100) - 1.25 * dDEF) * (100/dARC)  
  for i = HP, SIL do
    local singleStat = ds:GetSingleStat(i)
    local battleMod = singleStat:Get(BattleMod)
    if battleMod > 1 and Roll(0, 99) < 40 then
      a:BuffStat(i, battleMod - 1)
      singleStat:Set(BattleMod, 1)
      ds:SetSingleStat(singleStat, i)
    end
  end
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/SecretTechnique76.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Helper;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class SecretTechnique76 : SpellScriptBase
    {
        public override string Name { get { return "Secret Technique #76"; } }
        public override string Description { get { return "A vicious strike that attempts to weaken the enemy.\nChance to steal stat buffs from target."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.ARC, }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 76; } }
        public override double Delay { get { return .3; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (3.75 * aAD * (aARC / 100) - 1.25 * dDEF) * (100 / dARC);
            DealDamage(damage);
            for (var i = Stat.HP; i <= Stat.SIL; ++i)
            {
                var stat = target.Stats[i];
                var battleMod = stat[StatType.BattleMod];
                if (battleMod > 1 && RHelper.RollPercentage(40))
                {
                    BuffStat(caster, i, battleMod - 1);
                    stat[StatType.BattleMod] = 1;
                }
            }
        }
    }
}